### PR TITLE
Topic/ucs config

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**

### CODE REVIEW

File: src/core/xccl_init.c [hunk 1]
```diff
@@ -229,5 +231,63 @@ xccl_status_t xccl_init(const xccl_params_t *params,
     if (XCCL_OK != (status = xccl_lib_init(params, &lib))) {
         return status;
     }
+    ucs_config_parser_warn_unused_env_vars_once("XCCL_");
     return xccl_create_context(lib, config, context_p);
 }
+
+static ucs_config_field_t xccl_config_table[] = {
+  {"TEAMS", "all",
```
Comment (vspetrov) [inline src/core/xccl_init.c:239]: team "hier" - hierarchical

File: test/test_mpi.c [hunk 2]
```diff
@@ -35,12 +35,16 @@ static int oob_allgather(void *sbuf, void *rbuf, size_t msglen,
...
     int rank, size;
+    xccl_config_t *test_conf;
 
     MPI_Init(&argc, &argv);
     MPI_Comm_rank(MPI_COMM_WORLD, &rank);
     MPI_Comm_size(MPI_COMM_WORLD, &size);
 
+
     /* Init xccl library */
+    xccl_config_read(NULL, NULL, &test_conf);
```
Comment (vspetrov) [inline test/test_mpi.c:46]: Should this be xccl_config_read ("TEST", NULL, &test_conf) and then the following getenv can be removed?

File: src/core/xccl_init.c [hunk 3]
```diff
@@ -35,20 +56,20 @@ static int
...
 {
     char *str;
-    xccl_lib_t *lib = (xccl_lib_t*)data;
     if (NULL != (str = strstr(info->dlpi_name, "libxccl.so"))) {
         int pos = (int)(str - info->dlpi_name);
-        lib->lib_path = (char*)malloc(pos+8);
-        strncpy(lib->lib_path, info->dlpi_name, pos);
-        lib->lib_path[pos] = '\0';
-        strcat(lib->lib_path, "xccl");
+        free(xccl_lib_global_config.team_lib_path);
```
Comment (vspetrov) [inline src/core/xccl_init.c:63]: plz make sure this is a valid "free" on the first call of the "callback". It should contain "" empty string on that first call if i understand correctly - was it malloced by the ucs config parser?

**Review by vspetrov — APPROVED** [review]

File: src/core/xccl_init.c (hunk 1 above)
Comment (Sergei-Lebedev) [reply src/core/xccl_init.c:239]: done

File: test/test_mpi.c (hunk 2 above)
Comment (Sergei-Lebedev) [reply test/test_mpi.c:46]: Will be a part of another PR together with new param, config structures

File: src/core/xccl_init.c (hunk 3 above)
Comment (Sergei-Lebedev) [reply src/core/xccl_init.c:63]: ucs config assigns default value using strdup which in turn allocates memory, so this free is valid

**Pull request merged**